src/components/PlaylistPage/PlaylistPage.tsx

```diff
@@ -23,32 +24,44 @@ const PlaylistPage = () => {
     const { id } = useParams<{ id: string }>();
     const currentPlaylist = useSelector(selectCurrentPlaylist);
     const currentUser = useSelector(selectCurrentUser);
-    const [addSongActive, setAddSongActive] = useState(false);
-    const [inviteActive, setInviteActive] = useState(false);
-    const [playerActive, setPlayerActive] = useState(false);
-    const [confirmationIsVisible, setConfirmationIsVisible] = useState(false)
+    const [addSongActive, setAddSongActive] = useState<boolean>(false);
+    const [inviteActive, setInviteActive] = useState<boolean>(false);
+    const [playerActive, setPlayerActive] = useState<boolean>(false);
+    const [confirmationIsVisible, setConfirmationIsVisible] = useState<boolean>(false)
+    const [confirmationText, setConfirmationText] = useState<string>("")
     const isOwner = currentUser?.name === currentPlaylist?.ownerName;
     const dispatch = useDispatch();
     const history = useHistory();
 
-
-    const unFollow = async () => {
-        await dispatch(playlistsAsyncActions.unfollowPlaylist(id))
-        history.push('/')
+    const unFollowOnClick = () => {
+        setConfirmationIsVisible(true)
+        setConfirmationText("unfollow the playlist")
     }
 
-    const unFollowOnClick = () => {
+    const deleteOnClick = () => {
         setConfirmationIsVisible(true)
+        setConfirmationText("delete the playlist")
     }
 
     const confirmAction = (isConfirmed: boolean) => {
-        if (isConfirmed){
+        if (isConfirmed && confirmationText.includes("unfollow")){
```

Comment:
this solution seems wrong to me, maybe add a type to the confirmation?

Reply:
fixed